core/src/test/java/org/elasticsearch/bootstrap/BootstrapChecksTests.java

```diff
@@ -189,19 +188,15 @@ public void testFileDescriptorLimits() throws NodeValidationException {
         final AtomicLong maxFileDescriptorCount = new AtomicLong(randomIntBetween(1, limit - 1));
         final BootstrapChecks.FileDescriptorCheck check;
         if (osX) {
-            check = new BootstrapChecks.OsXFileDescriptorCheck() {
-                @Override
-                long getMaxFileDescriptorCount() {
-                    return maxFileDescriptorCount.get();
-                }
-            };
+            check =
+                    () -> {
+                        return maxFileDescriptorCount.get();
+                    };
```

Comment:
This can not possibly compile. There's no way the JVM can infer that the lambda here is intended override `BootstrapChecks.OsXFileDescriptorCheck`. Even if there were a way to infer it, I would prefer this as-is because as-is it's clear what method is being overridden, losing that behind lambda makes it less clear.